nikola/plugins/compile/rest/post_list.py

```diff
@@ -246,18 +251,16 @@ def _do_post_list(start=None, stop=None, reverse=False, tags=None, categories=No
     if sections:
         timeline = [p for p in timeline if p.section_name(lang).lower() in sections]
 
-    for post in timeline:
-        if tags:
-            cont = True
-            tags_lower = [t.lower() for t in post.tags]
-            for tag in tags:
-                if tag in tags_lower:
-                    cont = False
-
-            if cont:
-                continue
-
-        filtered_timeline.append(post)
+    if tags:
+        tags = {t.strip().lower() for t in tags.split(',')}
+        if require_all_tags:
+            compare = lambda x, y: x.issubset(y)
+        else:
+            compare = lambda x, y: x & y
+        for post in timeline:
+            post_tags = {t.lower() for t in post.tags}
+            if compare(tags, post_tags):
+                filtered_timeline.append(post)
```

Comment:
That’s not what I’m talking about. Read the old code: if no tags were specified, `filtered_timeline` would just be filled with all posts from `timeline`, otherwise there is a way for `continue` to fire and not put a post in the timeline. In this code, `filtered_timeline` may be appended to only if there are tags specified. But if there are no tags specified, the entire block does not fire, and `filtered_timeline` stays empty. After the tags block, `filtered_timeline` is used in place of `timeline`. So, to reiterate: `if not tags: filtered_timeline == []` and post list is *empty*.